packages/mgt/src/graph/graph.user.ts

```diff
@@ -123,11 +128,21 @@ export async function getUserWithPhoto(graph: IGraph, userId?: string): Promise<
     }
   }
   if (photosCacheEnabled()) {
-    const cachedPhoto = await getPhotoFromCache(userId || 'me', 'users');
+    cachedPhoto = await getPhotoFromCache(userId || 'me', 'users');
     if (cachedPhoto && getPhotoInvalidationTime() > Date.now() - cachedPhoto.timeCached) {
       photo = cachedPhoto.photo;
+    } else if (cachedPhoto) {
+      try {
+        const response = await graph.api(`${resource}/photo`).get();
+        if (response && response['@odata.mediaEtag'] === cachedPhoto.eTag) {
```

Comment:
would this still work as expected if `response['@odata.mediaEtag']` and `cachedPhoto.eTag` are both null?